proxy/api/protocol.go

```diff
@@ -28,6 +29,18 @@ type header struct {
 	flags  uint32
 }
 
+const maxPayloadLength = 1 * 1024 * 1024
```

Comment:
@amshinde could you take a look to @dlespiau  comment ?  If you agreed give a approval :)